### PR TITLE
Add development mode icons for browser extension

### LINKED ISSUE #34254: Research and create a signup/registration flow diagrams for cloud/on-prem/oss modes
> @[erzhan.torokulov@sourcegraph.com](mailto:erzhan.torokulov@sourcegraph.com) The scope here is probably just to tackle the new website visitor - > registration flow, but I want to make sure I also understand how new paid users are added to existing accounts. So...
>
> Can you also make sure the figma diagrams cover how new users are generated on existing paid accounts. If an account has 100 users and adds 10 more licenses - how do those users get:
> 1) Created on the back end when an admin adds them
> 2) Notified of their access
> 3) Set up login credentials - is this already set up by admin through SSO or is there also a need for some users to create passwords after they've been invited?

> Create a flow of current signup flow and desired with email validation flow

> Notes on data being shared (with other systems & also events sent to Amplitude/Looker) and the process behind each step

### PR DESCRIPTION
**Pull request by thenamankumar**
closes https://github.com/sourcegraph/sourcegraph/issues/33587

Changes browser extension icons in dev mode.

## Test plan

- Run extension `yarn --cwd client/browser run dev:chrome`
- Check extension icon
- Set extension to inactive and check icon again

## App preview:

- [Web](https://sg-web-naman-browser-extension-dev-icons.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-qnalznnuck.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/browser/src/browser-extension/browser-action-icon.ts [hunk 1]
```diff
@@ -16,23 +18,43 @@ interface BrowserActionIconPaths {
...
-        '48': 'img/icon-active-with-alert-48.png',
-        '128': 'img/icon-active-with-alert-128.png',
-    },
-    inactive: {
-        '32': 'img/icon-inactive-32.png',
-        '48': 'img/icon-inactive-48.png',
-        '128': 'img/icon-inactive-128.png',
-    },
-}
+const IsProductionVersion = !getExtensionVersion().startsWith('0.0.0')
```
Comment (erzhtor) [inline client/browser/src/browser-extension/browser-action-icon.ts:21]: WDYT to move to a shared place to reuse, like in https://github.com/sourcegraph/sourcegraph/blob/46b194ba527efb3205440f2acb63c5b837a1285e/client/browser/src/browser-extension/scripts/backgroundPage.main.ts#L53

**Review by erzhtor — APPROVED** [review]

File: client/browser/src/browser-extension/browser-action-icon.ts (hunk 1 above)
Comment (thenamankumar) [reply client/browser/src/browser-extension/browser-action-icon.ts:21]: can do, will do.

**Comment by erzhtor** [comment]
Also, I forgot to mention a Browser Extension CHANGELOG.md file. I think it is worth noting this change there, too, maybe. cc @taras-yemets

File: client/browser/src/browser-extension/browser-action-icon.ts (hunk 1 above)
Comment (thenamankumar) [reply client/browser/src/browser-extension/browser-action-icon.ts]: done

**Pull request merged**